### ISSUE TITLE
Add containsBy to RichIterable

### ISSUE THREAD

**Issue by donraab** [eclipse/eclipse-collections#919]
Occasionally, developers want to do the following:

`collection.asLazy().collect(Type::someAttribute).contains(value);`

I've seen this quite often written in the equivalent form:

`collection.anySatisfy(type -> type.someAttribute().equals(value));`

Both of these solutions could be more simply written as a fused method:

`collection.collectContains(Type::someAttribute, value);`

or 

`collection.contains(Type::someAttribute, value);`

update with new name: 

`collection.containsBy(Type::someAttribute, value);`

**Comment by donraab** [comment]
@motlin @nikhilnanivadekar Thoughts on this? I found this pattern in exercise 8 solutions of the Company Kata.

**Comment by KharcheNikhil** [comment]
If this is accepted, I would like to contribute. In fact have coded the change with one of the recommended method name in my repository.

**Comment by motlin** [comment]
I've used the original pattern many times:
```
collection.asLazy().collect(Type::someAttribute).contains(value);
```
Seems fine to me.

**Comment by donraab** [comment]
Hi @KharcheNikhil, I wrote a blog about this [feature on Medium](https://medium.com/@donraab/fusing-methods-for-productivity-c15c9eb2d666?source=friends_link&sk=ddde162c6aab068bda4619266d3b3bfb) today and got a suggestion from @vmzakharov about a better method name. I had already implemented most of the code and will likely submit a PR once I have properly written all of the tests.

Thank you for the offer to contribute this and hopefully this did not take a lot of effort on your part. I should not have put the help wanted and good first issue tags as had already started implementing after I wrote the issue. Have a look at the blog and hopefully you'll see why I wanted to write this. I wanted to show the evolution of and decision making behind adding a fused method. This was a natural follow up to my last blog on moving [From Eager to Fused to Lazy](https://medium.com/javarevisited/from-eager-to-fused-to-lazy-66450e290002?source=friends_link&sk=1c61f413a8148ece8e60e0a2a87137b3).
